### ISSUE TITLE
error: Handle check failed: saw Function '_FfiCallbacksimpleAddition': static ffi-trampoline-function. expected Instance

### ISSUE THREAD

**Issue by a-siva** [dart-lang/sdk#40967]
The following tests crash when running in NNBD strong mode

```
dartk-strong-linux-release-x64:ffi/function_callbacks_many_test/0 is new and failed (Crash, expected Pass)
dartk-strong-linux-release-x64:ffi/function_callbacks_many_test/1 is new and failed (Crash, expected Pass)
dartk-strong-linux-release-x64:ffi/function_callbacks_many_test/2 is new and failed (Crash, expected Pass)
dartk-strong-linux-release-x64:ffi/function_callbacks_many_test/3 is new and failed (Crash, expected Pass)
dartk-strong-linux-release-x64:ffi/function_callbacks_many_test/4 is new and failed (Crash, expected Pass)
dartk-strong-linux-release-x64:ffi/function_callbacks_many_test/5 is new and failed (Crash, expected Pass)
dartk-strong-linux-release-x64:ffi/function_callbacks_many_test/6 is new and failed (Crash, expected Pass)
dartk-strong-linux-release-x64:ffi/function_callbacks_many_test/7 is new and failed (Crash, expected Pass)
dartk-strong-linux-release-x64:ffi/function_callbacks_test/0 is new and failed (Crash, expected Pass)
dartk-strong-linux-release-x64:ffi/function_callbacks_test/1 is new and failed (Crash, expected Pass)
dartk-strong-linux-release-x64:ffi/function_callbacks_test/2 is new and failed (Crash, expected Pass)
dartk-strong-linux-release-x64:ffi/function_callbacks_test/3 is new and failed (Crash, expected Pass)
dartk-strong-linux-release-x64:ffi/function_callbacks_test/4 is new and failed (Crash, expected Pass)
dartk-strong-linux-release-x64:ffi/function_callbacks_test/5 is new and failed (Crash, expected Pass)
dartk-strong-linux-release-x64:ffi/function_callbacks_test/6 is new and failed (Crash, expected Pass)
dartk-strong-linux-release-x64:ffi/function_callbacks_test/7 is new and failed (Crash, expected Pass)
dartk-strong-linux-release-x64:ffi/function_callbacks_very_many_test/0 is new and failed (Crash, expected Pass)
dartk-strong-linux-release-x64:ffi/function_callbacks_very_many_test/1 is new and failed (Crash, expected Pass)
dartk-strong-linux-release-x64:ffi/function_callbacks_very_many_test/2 is new and failed (Crash, expected Pass)
dartk-strong-linux-release-x64:ffi/function_callbacks_very_many_test/3 is new and failed (Crash, expected Pass)
dartk-strong-linux-release-x64:ffi/function_callbacks_very_many_test/4 is new and failed (Crash, expected Pass)
dartk-strong-linux-release-x64:ffi/function_callbacks_very_many_test/5 is new and failed (Crash, expected Pass)
dartk-strong-linux-release-x64:ffi/function_callbacks_very_many_test/6 is new and failed (Crash, expected Pass)
dartk-strong-linux-release-x64:ffi/function_callbacks_very_many_test/7 is new and failed (Crash, expected Pass)
dartk-strong-linux-release-x64:ffi/function_callbacks_very_many_test/8 is new and failed (Crash, expected Pass)
dartk-strong-linux-release-x64:ffi/hardfp_test/0 is new and failed (Crash, expected Pass)
dartk-strong-linux-release-x64:ffi/hardfp_test/1 is new and failed (Crash, expected Pass)
dartk-strong-linux-release-x64:ffi/hardfp_test/2 is new and failed (Crash, expected Pass)
dartk-strong-linux-release-x64:ffi/hardfp_test/3 is new and failed (Crash, expected Pass)
dartk-strong-linux-release-x64:ffi/hardfp_test/4 is new and failed (Crash, expected Pass)
dartk-strong-linux-release-x64:ffi/hardfp_test/5 is new and failed (Crash, expected Pass)
dartk-strong-linux-release-x64:ffi/hardfp_test/6 is new and failed (Crash, expected Pass)
dartk-strong-linux-release-x64:ffi/stacktrace_regress_37910_test is new and failed (Crash, expected Pass)
dartk-strong-linux-release-x64:ffi/variance_function_test/0 is new and failed (Crash, expected Pass)
dartk-strong-linux-release-x64:ffi/variance_function_test/1 is new and failed (Crash, expected Pass)
dartk-strong-linux-release-x64:ffi/variance_function_test/2 is new and failed (Crash, expected Pass)
dartk-strong-linux-release-x64:ffi/vmspecific_function_callbacks_exit_test/01/0 is new and failed (Crash, expected Pass)
dartk-strong-linux-release-x64:ffi/vmspecific_function_callbacks_exit_test/01/1 is new and failed (Crash, expected Pass)
dartk-strong-linux-release-x64:ffi/vmspecific_function_callbacks_exit_test/02/0 is new and failed (Crash, expected Pass)
dartk-strong-linux-release-x64:ffi/vmspecific_function_callbacks_exit_test/02/1 is new and failed (Crash, expected Pass)
dartk-strong-linux-release-x64:ffi/vmspecific_function_callbacks_exit_test/03/0 is new and failed (Crash, expected Pass)
dartk-strong-linux-release-x64:ffi/vmspecific_function_callbacks_exit_test/03/1 is new and failed (Crash, expected Pass)
dartk-strong-linux-release-x64:ffi/vmspecific_function_callbacks_test/0 is new and failed (Crash, expected Pass)
dartk-strong-linux-release-x64:ffi/vmspecific_function_callbacks_test/1 is new and failed (Crash, expected Pass)
dartk-strong-linux-release-x64:ffi/vmspecific_function_callbacks_test/2 is new and failed (Crash, expected Pass)
dartk-strong-linux-release-x64:ffi/vmspecific_function_callbacks_test/3 is new and failed (Crash, expected Pass)
dartk-strong-linux-release-x64:ffi/vmspecific_function_callbacks_test/4 is new and failed (Crash, expected Pass)
dartk-strong-linux-release-x64:ffi/vmspecific_function_callbacks_test/5 is new and failed (Crash, expected Pass)
dartk-strong-linux-release-x64:ffi/vmspecific_function_callbacks_test/6 is new and failed (Crash, expected Pass)
dartk-strong-linux-release-x64:ffi/vmspecific_function_callbacks_test/7 is new and failed (Crash, expected Pass)
dartk-strong-linux-release-x64:ffi/vmspecific_function_callbacks_test/8 is new and failed (Crash, expected Pass)
dartk-strong-linux-release-x64:ffi/vmspecific_function_callbacks_test/9 is new and failed (Crash, expected Pass)
dartk-strong-linux-release-x64:ffi/vmspecific_regress_37511_callbacks_test is new and failed (Crash, expected Pass)

```
```
--- Command "vm" (took 09.000976s):
DART_CONFIGURATION=ReleaseX64NNBD out/ReleaseX64NNBD/dart --enable-testing-pragmas --null-safety --enable-experiment=non-nullable --ignore-unrecognized-flags --packages=/b/s/w/ir/cache/builder/sdk/.packages /b/s/w/ir/cache/builder/sdk/tests/ffi/function_callbacks_many_test.dart

exit code:
-6

stderr:
../../runtime/vm/object.h: 7111: error: Handle check failed: saw Function '_FfiCallbacksimpleAddition': static ffi-trampoline-function. expected Instance
version=2.8.0-edge.24480f24b6cd04897111ca78716b6d4fa980bb3a (Tue Mar 10 23:29:18 2020 +0000) on "linux_x64"
thread=9586, isolate=main(0x55adfbdd8900)
isolate_instructions=55adfa061620, vm_instructions=55adfa061620
  pc 0x000055adfa32e89c fp 0x00007f1e5a87d3c0 dart::Profiler::DumpStackTrace(void*)
  pc 0x000055adfa061752 fp 0x00007f1e5a87d4a0 dart::Assert::Fail(char const*, ...)
  pc 0x000055adfa1e0758 fp 0x00007f1e5a87d4d0 out/ReleaseX64NNBD/dart+0x1a24758
  pc 0x000055adfa3790a3 fp 0x00007f1e5a87d690 dart::DRT_TypeCheck(dart::NativeArguments)
  pc 0x00007f1e5b981108 fp 0x00007f1e5a87d6d0 Unknown symbol
  pc 0x00007f1e5a9b7951 fp 0x00007f1e5a87d758 Unknown symbol
  pc 0x00007f1e5a9b777e fp 0x00007f1e5a87d7a0 Unknown symbol
  pc 0x00007f1e5a9b7673 fp 0x00007f1e5a87d7d0 Unknown symbol
  pc 0x00007f1e5a9ab528 fp 0x00007f1e5a87d818 Unknown symbol
  pc 0x00007f1e5a9ab3c3 fp 0x00007f1e5a87d840 Unknown symbol
  pc 0x00007f1e5a9ab317 fp 0x00007f1e5a87d880 Unknown symbol
  pc 0x00007f1e5a9ab157 fp 0x00007f1e5a87d8b8 Unknown symbol
  pc 0x00007f1e5a9aa47c fp 0x00007f1e5a87d900 Unknown symbol
  pc 0x00007f1e5a9aa18a fp 0x00007f1e5a87d940 Unknown symbol
  pc 0x00007f1e5a9aa09e fp 0x00007f1e5a87d978 Unknown symbol
  pc 0x00007f1e5b98159c fp 0x00007f1e5a87d9e8 Unknown symbol
  pc 0x000055adfa2456be fp 0x00007f1e5a87da90 dart::DartEntry::InvokeFunction(dart::Function const&, dart::Array const&, dart::Array const&, unsigned long)
  pc 0x000055adfa2476a3 fp 0x00007f1e5a87dae0 dart::DartLibraryCalls::HandleMessage(dart::Object const&, dart::Instance const&)
  pc 0x000055adfa27f5b6 fp 0x00007f1e5a87dcd0 dart::IsolateMessageHandler::HandleMessage(std::__2::unique_ptr<dart::Message, std::__2::default_delete<dart::Message> >)
  pc 0x000055adfa2aafe6 fp 0x00007f1e5a87dd40 dart::MessageHandler::HandleMessages(dart::MonitorLocker*, bool, bool)
  pc 0x000055adfa2ab7d0 fp 0x00007f1e5a87dda0 dart::MessageHandler::TaskCallback()
  pc 0x000055adfa3abb10 fp 0x00007f1e5a87dde0 dart::ThreadPool::Worker::Loop()
  pc 0x000055adfa3ab98e fp 0x00007f1e5a87de10 dart::ThreadPool::Worker::Main(unsigned long)
  pc 0x000055adfa32af86 fp 0x00007f1e5a87ded0 out/ReleaseX64NNBD/dart+0x1b6ef86
-- End of DumpStackTrace
[exit     : sp(0) fp(0x7f1e5a87d6d0) pc(0)]
[dart     : sp(0x7f1e5a87d6e0) fp(0x7f1e5a87d758) pc(0x7f1e5a9b7951) dart:ffi_::__nativeCallbackFunction@8050071 ]
[dart     : sp(0x7f1e5a87d768) fp(0x7f1e5a87d7a0) pc(0x7f1e5a9b777e) file:///b/s/w/ir/cache/builder/sdk/tests/ffi/function_callbacks_many_test.dart_::_init__#ffiCallback0@19387306 ]
[dart     : sp(0x7f1e5a87d7b0) fp(0x7f1e5a87d7d0) pc(0x7f1e5a9b7673) file:///b/s/w/ir/cache/builder/sdk/tests/ffi/function_callbacks_many_test.dart_::_get__#ffiCallback0@19387306 ]
[dart     : sp(0x7f1e5a87d7e0) fp(0x7f1e5a87d818) pc(0x7f1e5a9ab528) file:///b/s/w/ir/cache/builder/sdk/tests/ffi/function_callbacks_many_test.dart_::_testManyCallbacks ]
[dart     : sp(0x7f1e5a87d828) fp(0x7f1e5a87d840) pc(0x7f1e5a9ab3c3) file:///b/s/w/ir/cache/builder/sdk/tests/ffi/function_callbacks_many_test.dart_::_main ]
[dart     : sp(0x7f1e5a87d850) fp(0x7f1e5a87d880) pc(0x7f1e5a9ab317) file:///b/s/w/ir/cache/builder/sdk/tests/ffi/function_callbacks_many_test.dart_::_main_main ]
[dart     : sp(0x7f1e5a87d890) fp(0x7f1e5a87d8b8) pc(0x7f1e5a9ab157) dart:core__Closure@0150898_call ]
[dart     : sp(0x7f1e5a87d8c8) fp(0x7f1e5a87d900) pc(0x7f1e5a9aa47c) dart:isolate_::__startIsolate@1026248_<anonymous closure> ]
[dart     : sp(0x7f1e5a87d910) fp(0x7f1e5a87d940) pc(0x7f1e5a9aa18a) dart:core__Closure@0150898_call ]
[dart     : sp(0x7f1e5a87d950) fp(0x7f1e5a87d978) pc(0x7f1e5a9aa09e) dart:isolate__RawReceivePortImpl@1026248__handleMessage@1026248 ]
[entry    : sp(0x7f1e5a87d988) fp(0x7f1e5a87d9e8) pc(0x7f1e5b98159c)]

```

**Comment by dcharkes** [comment]
@a-siva just to double check:

* This does not block unforking the SDK? (Because that runs in legacy mode?)
* This landed and the failures are approved?

**Comment by a-siva** [comment]
Yes this does not block unforking, I have turned the testing on and have approved these failures.

**Comment by alexmarkov** [comment]
These crashes were fixed in 65a97ae1fa772e15d4b4a950989546c7a157c5a0.
